actors/builtin/miner/sectors.go

```diff
@@ -0,0 +1,89 @@
+package miner
+
+import (
+	"fmt"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
+
+	"github.com/ipfs/go-cid"
+)
+
+func LoadSectors(store adt.Store, root cid.Cid) (*Sectors, error) {
+	sectorsArr, err := adt.AsArray(store, root)
+	if err != nil {
+		return nil, err
+	}
+	return (*Sectors)(sectorsArr), nil
+}
+
+// Sectors is a helper type for accessing/modifying a miner's sectors. It's safe
+// to pass this object around as needed.
+type Sectors adt.Array
```

Comment:
Discussion: here you opted to _newtype_ Sectors as an immediate (not pointer) Array, and then generally use pointer to sectors. In the BitfieldQueue and ExpirationQueue I opted to _extend_ Array _pointer_, and then treat the queue as an immediate/value type, generally not returned as or passed by pointer. (I subsequently added QuantSpec to it, so +1 for extensibility but -1 for being less lightweight as an immediate).

I don't think I have sufficiently deep Go experience to fully weigh these two approaches against each other, but it occurs that if one way is clearly better then we should use it everywhere, and if we're ambivalent then we should just pick one and use it consistently. 

Can you speak to the benefits of one approach over the other?

Reply:
I used a new type in this case because I wanted to expose functions like `Get` and `MustGet` and didn't want them to be confused with the array functions. But it doesn't really matter much either way.

In general, I try to avoid passing around a `type Foo struct { *Bar }` by value. If I then decide that I really _need_ an additional mutable field, I'll need to refactor everything to pass it around by pointer. Instead, I prefer `type Foo Bar` if I don't want to expose the underlying functions and `type Foo struct { Bar }` If I can get away with the dereference (i.e., no locks).

Otherwise, I just eat the pointer indirection and pass `type Foo struct { Bar }` by pointer.

Reply:
I've switched over to `type Foo struct { *Bar }`. We can change it later if we want but I agree that consistency is king.

Reply:
Thanks, your rationale sounds good. Thanks for the consistency, I'll be happy for us to switch them all to `struct { Bar }